text/0000-minimal-builders.md

```diff
@@ -0,0 +1,246 @@
+onto the test runner's daemon – and then all of the pulled buildpacks are
+promptly ignored. Pulling unnecessary layers slows down Paketo's tests.
+
+This RFC proposes that Paketo publish builders that package up build- and
+run-images with the lifecycle but exclude buildpacks and orders. These builders
+would be significantly smaller than the existing builders, leading to shorter
+builder image pull times.
+
+Since the CNB spec currently [supports including `[[ buildpacks ]]` in a
+project file
+(`project.toml`)](https://buildpacks.io/docs/reference/config/project-descriptor/),
+users can specify the buildpack(s) for their build in an easily stored and
+reproducible way. Users can, of course, also specify which buildpacks to
+include in the build with the `--buildpack` flag for pack.
+
+The proposed new builders will thus speed up users' builds and Paketo's
+buildpack tests while relying only upon the existing functionality that the
+pack CLI provides.(No upstream contributions required.)
+
+## Detailed Explanation
+
+### Definitions
+- **minimal builder** : A
+  [builder](https://buildpacks.io/docs/concepts/components/builder/) that
+  includes a build-image, a reference to a run-image, and a lifecycle, _but no
+  buildpacks nor order groups_.
+- **standard builder**: A builder that includes buildpacks and order groups in
+  addition to the build-image, lifecycle, and reference to a run-image.
+
+### Calculating Builder Size
+
+To determine the number of layers in a given builder image, one can use the
+`docker history` command. Its output shows one line per layer. For example, to
+calculate the number of layers in the latest Paketo Full builder, simply run:
+```
+docker pull paketobuildpacks/builder:full
+docker history -q paketobuildpacks/builder:full | wc -l
+```
+
+### What will the Paketo Minimal Builders look like?
+The `builder.toml` for a given Paketo Minimal <Name> Builder will be identical
+to the corresponding standard Paketo <Name> Builder, but will have the `[[
+buildpacks ]]` and `[[ order ]]` sections removed. For instance, the Minimal
+Full Builder will have a `builder.toml` that looks like:
+
+```
+description = "<some description that clearly states buildpacks must be specified with this builder>"
+
+[lifecycle]
+  version = "X.Y.Z"
+
+[stack]
+  build-image = "docker.io/paketobuildpacks/build:X.Y.Z-full-cnb"
+  id = "io.buildpacks.stacks.bionic"
+  run-image = "index.docker.io/paketobuildpacks/run:full-cnb"
+  run-image-mirrors = ["gcr.io/paketo-buildpacks/run:full-cnb"]
+```
+
+The above stated change results in a ~88% decrease in the number of layers
+as compared to Paketo Full Builder v0.1.130.
+
+### How will the Paketo Minimal Builders be distributed?
+Paketo will maintain and publish minimal builders that correspond to each of
+its three existing standard builders. Their config files (`builder.toml`) will
+be checked in to their own repos (paketo-buildpacks/minimal-full-builder,
```

Comment:
Any need to maintain them separately if one is a subset of the other?
How about for e.g. the full-builder also publishes the full-builder-minimal with the same version every time it is published, but minus [order]?

Reply:
I'm not opposed to this. Definitely reduces repo overhead. I wonder if it'd be confusing for the version to be bumped with no change to the actual builder (in the event of a buildpack update).

Reply:
I switched the RFC to this approach.